rxjava-reactive-streams/src/main/java/rx/internal/reactivestreams/SingleAsPublisher.java

```diff
@@ -16,56 +16,55 @@
 
 package rx.internal.reactivestreams;
 
-import java.util.concurrent.atomic.AtomicInteger;
-
 import org.reactivestreams.Publisher;
 import org.reactivestreams.Subscriber;
 import org.reactivestreams.Subscription;
+import rx.Single;
+import rx.SingleSubscriber;
 
-import rx.*;
+import java.util.concurrent.atomic.AtomicInteger;
 
 /**
  * Wraps a Single and exposes it as a Publisher.
  *
  * @param <T> the value type
  */
 public final class SingleAsPublisher<T> implements Publisher<T> {
-    
+
     final Single<T> single;
-    
+
     public SingleAsPublisher(Single<T> single) {
         this.single = single;
     }
 
     @Override
     public void subscribe(Subscriber<? super T> s) {
         SingleAsPublisherSubscriber<T> parent = new SingleAsPublisherSubscriber<T>(s);
-        s.onSubscribe(parent);
-        
         single.subscribe(parent);
+        s.onSubscribe(parent);
```

Comment:
This  violates the RS contract. OnSubscribe must happen before an other calls to onXXX.